R/survey_weights.R

```diff
@@ -0,0 +1,227 @@
+#'   "Village A",             1,            6,           4,              4,  "50-60",  "F",      "N",
+#'   "Village A",             2,            6,           4,              4,  "50-60",  "F",      "N",
+#'   "Village B",             2,            3,           3,              1,  "20-30",  "M",      "N",
+#'   "Village B",             2,            3,           3,              2,  "50-60",  "F",      "N",
+#'   "Village B",             2,            3,           3,              3,  "30-40",  "F",      "Y"
+#' )
+#'
+#'
+#' cz <- tibble::tribble(
+#'   ~cluster, ~n_houses,
+#'   "Village A",        23,
+#'   "Village B",        42,
+#'   "Village C",        56,
+#'   "Village D",        38
+#' )
+#'
+
+
+add_weights <- function(x, p, cz = NULL, ... , population,
+                        cluster_cz, household_cz,
+                        cluster_x, household_x, individuals_eligible_x, individuals_interviewed_x,
+                        method = "stratified",
+                        ignore_cluster = TRUE, ignore_household = TRUE,
+                        surv_weight = "surv_weight", surv_weight_ID =  "surv_weight_ID") {
+
+  # stratified method is straight forward - lifted directly from old function version
+  if (method == "stratified") {
+    .dots      <- rlang::enquos(...)
+    population <- rlang::enquo(population)
+    surv_weight_ID  <- rlang::enquo(surv_weight_ID)
+    surv_weight     <- rlang::enquo(surv_weight)
+
+    # create a merger ID by age group and sex
+    p <- tidyr::unite(p, "ID", !!! .dots)
+
+    # get study sample counts
+    counts <- dplyr::count(x, !!! .dots)
+    counts <- tidyr::unite(counts, "ID", !!! .dots)
+    counts <- dplyr::select(counts, .data$ID, .data$n)
+
+    # merge counts to population data
+    p <- dplyr::left_join(p, counts, by = "ID")
+
+    # create weight variable
+    p <- dplyr::mutate(p, !! surv_weight := !! population / .data$n)
+    p <- dplyr::select(p, .data$ID, !! surv_weight)
+
+
+    # merge to study sample
+    merge_by <- "ID"
+    names(merge_by) <- as.character(rlang::expr(surv_weight_ID))
+    d <- tidyr::unite(x, !! surv_weight_ID, !!! .dots, remove = FALSE)
+    d <- dplyr::left_join(d, p, by = merge_by)
+    d
+  }  else {
+
+    # cluster method
+
+    # define vars
+    clus_id_cz <- tidyselect::vars_select(colnames(cz), {{cluster_cz}})
+    hh_id_cz   <- tidyselect::vars_select(colnames(cz), {{household_cz}})
+    clus_id_x <- tidyselect::vars_select(colnames(x), {{cluster_x}})
+    hh_id_x   <- tidyselect::vars_select(colnames(x), {{household_x}})
+    indiv_eli   <- tidyselect::vars_select(colnames(x), {{individuals_eligible_x}})
+    indiv_surv   <- tidyselect::vars_select(colnames(x), {{individuals_interviewed_x}})
+
+
+    if (ignore_cluster) {
+      cluster_chance <- 1
+    } else {
+      ## chance of a cluster being selected
+      # the total number of clusters in the area
```

Comment:
you don't need data frames for this. You are comparing the number of unique clusters in the area vs the number of clusters surveyed and creating a proportion out of those? 

```r
n_clusters_total <- length(unique(cs[[clus_id_cz]]))
n_clusters_chosen <- length(unique(z[[clus_id_x]])
cluster_chance <- n_clusters_total / n_clusters_chosen
```